### PR TITLE
add support for mv cols with realtime lucene index

### PR DESCRIPTION
**Pull request by itschrispeck**
Adds support to build lucene text index in the mutable segment for MV cols

addresses: https://github.com/apache/pinot/issues/10975

### CODE REVIEW

File: pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/mutable/MutableTextIndex.java [hunk 1]
```diff
@@ -31,12 +31,18 @@ default void add(@Nonnull Object value, int dictId, int docId) {
...
    * Index the document
    * @param document the document as a string
    */
   void add(String document);
+
+  /**
+   * Index the multi-value document
+   * @param document the document as an array of Strings
+   */
+  void add(String[] document);
```
Comment (Jackie-Jiang) [inline pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/mutable/MutableTextIndex.java:47]: (minor) Same for other signatures
```suggestion
  void add(String[] documents);
```

File: pinot-segment-local/src/test/java/org/apache/pinot/segment/local/realtime/impl/invertedindex/LuceneMutableTextIndexTest.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import org.apache.commons.io.FileUtils;
+import org.apache.lucene.search.SearcherManager;
+import org.roaringbitmap.buffer.ImmutableRoaringBitmap;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+
+public class LuceneMutableTextIndexTest {
```
Comment (Jackie-Jiang) [inline pinot-segment-local/src/test/java/org/apache/pinot/segment/local/realtime/impl/invertedindex/LuceneMutableTextIndexTest.java:31]: Please reformat this class with [Pinot Style](https://docs.pinot.apache.org/developers/developers-and-contributors/code-setup#setup-ide)

**Review by Jackie-Jiang — COMMENTED** [review]
Thanks for the contribution!

File: pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/mutable/MutableTextIndex.java (hunk 1 above)
Comment (itschrispeck) [reply pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/mutable/MutableTextIndex.java]: thanks for the comments! addressed

**Review by Jackie-Jiang — APPROVED** [review]

**Pull request merged**